zerver/views/custom_profile_fields.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import absolute_import
+
+from typing import Text, Union, List, Dict
+
+from django.core.exceptions import ValidationError
+from django.db import IntegrityError
+from django.http import HttpRequest, HttpResponse
+from django.utils.translation import ugettext as _
+
+from zerver.decorator import has_request_variables, REQ, require_realm_admin
+from zerver.lib.actions import (do_add_realm_custom_profile_field,
+                                do_remove_realm_custom_profile_field,
+                                do_update_realm_custom_profile_field,
+                                do_update_user_custom_profile_data)
+from zerver.lib.response import json_success, json_error
+from zerver.lib.validator import check_dict, check_list, check_int, check_string
+
+from zerver.models import (custom_profile_fields_for_realm, UserProfile,
+                           CustomProfileField, custom_profile_fields_for_realm)
+
+def list_realm_custom_profile_fields(request, user_profile):
+    # type: (HttpRequest, UserProfile) -> HttpResponse
+    fields = custom_profile_fields_for_realm(user_profile.realm_id)
+    return json_success({'custom_fields': fields})
+
+@require_realm_admin
+@has_request_variables
+def create_realm_custom_profile_field(request, user_profile, name=REQ(),
+                                      field_type=REQ(validator=check_int)):
+    # type: (HttpRequest, UserProfile, Text, int) -> HttpResponse
+    try:
+        field_id = do_add_realm_custom_profile_field(
+            realm=user_profile.realm,
+            name=name,
+            field_type=field_type,
+        )
+        return json_success({'id': field_id})
+    except ValidationError as e:
+        return json_error(e.messages[0], data={"errors": dict(e)})
```

Comment:
This should check up front that the field type is one of the defined field types.